src/api/methods/define/Define.ts

```diff
@@ -100,3 +100,135 @@ export async function defineTransaction(session: AbstractSession, parms: ITransa
+ * @throws {ImperativeError} CICS CSD group not defined or blank
+ * @throws {ImperativeError} CICS URIMap path not defined or blank
+ * @throws {ImperativeError} CICS URIMap host not defined or blank
+ * @throws {ImperativeError} CICS URIMap scheme not defined or blank
+ * @throws {ImperativeError} CICS region name not defined or blank
+ * @throws {ImperativeError} CICS URIMap pipeline name not defined or blank
+ * @throws {ImperativeError} CicsCmciRestClient request fails
+ */
+export async function defineUrimapPipeline(session: AbstractSession, parms: IURIMapParms): Promise<ICMCIApiResponse> {
+    validateUrimapParms(parms);
+    ImperativeExpect.toBeDefinedAndNonBlank(parms.pipelineName, "CICS URIMap Pipeline name", "CICS URIMap pipeline name is required");
+
+    Logger.getAppLogger().debug("Attempting to define a pipeline URIMap with the following parameters:\n%s", JSON.stringify(parms));
+    const requestBody: any = buildUrimapRequestBody(parms, "pipeline");
+    requestBody.request.create.attributes.$.pipeline = parms.pipelineName;
+
+    const cicsPlex = parms.cicsPlex == null ? "" : parms.cicsPlex + "/";
+    const cmciResource = "/" + CicsCmciConstants.CICS_SYSTEM_MANAGEMENT + "/" +
+        CicsCmciConstants.CICS_DEFINITION_URIMAP + "/" + cicsPlex + parms.regionName;
+    return CicsCmciRestClient.postExpectParsedXml(session, cmciResource, [], requestBody) as any;
+}
+
+/**
+ * Validate parameters for defining a URIMap
+ * @param {IURIMapParms} parms - parameters to validate
+ * @throws {ImperativeError} CICS URIMap name not defined or blank
+ * @throws {ImperativeError} CICS CSD group not defined or blank
+ * @throws {ImperativeError} CICS URIMap path not defined or blank
+ * @throws {ImperativeError} CICS URIMap host not defined or blank
+ * @throws {ImperativeError} CICS URIMap scheme not defined or blank
+ * @throws {ImperativeError} CICS region name not defined or blank
+ */
+function validateUrimapParms(parms: IURIMapParms) {
+    ImperativeExpect.toBeDefinedAndNonBlank(parms.name, "CICS URIMap Name", "CICS URIMap name is required");
+    ImperativeExpect.toBeDefinedAndNonBlank(parms.csdGroup, "CICS CSD Group", "CICS CSD group is required");
+    ImperativeExpect.toBeDefinedAndNonBlank(parms.path, "CICS URIMap Path", "CICS URIMap path is required");
+    ImperativeExpect.toBeDefinedAndNonBlank(parms.host, "CICS URIMap Host", "CICS URIMap host is required");
+    ImperativeExpect.toBeDefinedAndNonBlank(parms.scheme, "CICS URIMap Scheme", "CICS URIMap scheme is required");
+    ImperativeExpect.toBeDefinedAndNonBlank(parms.regionName, "CICS Region name", "CICS region name is required");
+}
+
+/**
+ * Build request body for defining a URIMap
+ * @param {IURIMapParms} parms - parameters containing attribute values
+ * @param {string} usage - value of the usage attribute (server, client, or pipeline)
+ */
+function buildUrimapRequestBody(parms: IURIMapParms, usage: "server" | "client" | "pipeline") {
```

Comment:
Since this isn't an option that gets passed on the command line, I was leaning on the side of the fence against doing this :smile: Perhaps it would be best practice to still create one?